pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -17,22 +17,68 @@ limitations under the License.
 package util
 
 import (
+	"strings"
+
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/typed/discovery"
 	"k8s.io/kubernetes/pkg/kubectl"
 )
 
 // ShortcutExpander is a RESTMapper that can be used for OpenShift resources.   It expands the resource first, then invokes the wrapped
 type ShortcutExpander struct {
 	RESTMapper meta.RESTMapper
 
-	All []string
+	All []unversioned.GroupResource
+
+	discoveryClient *discovery.DiscoveryClient
 }
 
 var _ meta.RESTMapper = &ShortcutExpander{}
 
-func NewShortcutExpander(delegate meta.RESTMapper) ShortcutExpander {
-	return ShortcutExpander{All: userResources, RESTMapper: delegate}
+func NewShortcutExpander(delegate meta.RESTMapper, clients ...*discovery.DiscoveryClient) ShortcutExpander {
```

Comment:
i wanted to leave the existing instances of NewShortcutExpander with just one parameter as is since you mentioned "You could make discoveryClient optional" :)